### PR TITLE
subfig ToC handling moved to maine-thesis.cls

### PR DESCRIPTION
**Pull request by jcwhittier**
I've updated `maine-thesis.cls` and the documentation.

### CODE REVIEW

File: maine-thesis.cls [hunk 1]
```diff
@@ -336,6 +342,13 @@
 	#1
 }
 
+% When using the subfig package, make sure figures and tables are labeled correctly in the ToC 
+\AfterPackage{subfig}{ 
+	\captionsetup{labelsep=period,listofformat=simple}
+	\renewcommand\p@figure{Figure\space}
+	\renewcommand\p@table{Table\space}
+}
+
```
Comment (rpspringuel) [inline maine-thesis.cls]: Why are you doing this twice?

File: Documentation/AppA.tex [hunk 2]
```diff
@@ -26,19 +26,25 @@ \section{Working Packages}
...
+The one exception to this is if you use the caption or subfig package (which depends on the caption package) to create multi-page floats.\footnote{If you need to create subfigures but don't need a figure to span multiple pages, use the subfigure package, as it won't conflict with this class file.} If you do, you'll find that the caption formats and the List of Figures/Tables do not typeset the way the Graduate School wants them to be typeset.  In particular, the separator between ``Figure \#.\#'' and the caption will be ``:\space'' instead of ``.\space'' and the word ``Figure'' or ``Table'' will not accompany the number in the respective list.  To avoid this, the following code has been added to maine-thesis.cls:
+\begin{verbatim}
+\AfterPackage{subfig}{
+	\captionsetup{labelsep=period,listofformat=simple}
+	\renewcommand\p@figure{Figure\space}
+	\renewcommand\p@table{Table\space}
+}
+\end{verbatim}
```
Comment (rpspringuel) [inline Documentation/AppA.tex]: TMI.  The end user doesn't need to know what is happening behind the scenes, only the special actions they need to take when using these packages.

**Review by rpspringuel — CHANGES_REQUESTED** [review]
See individual comments.

**Comment by jcwhittier** [comment]
I didn't realize my earlier change persisted after I merged your changes.
Do you have a preference on which section it should be in? Keep 158 and
delete 345?

I will shorten the documentation.

**Comment by rpspringuel** [comment]
Definately the earlier location and in fact you might move it to before the `\ifPDFTeX` statement so that it directly follows the `\AfterPackage{hyperref}` block.   That way all the after package stuff is in the same place.

**Comment by jcwhittier** [comment]
This change causes conflict with the behavior of `\ref` and `\autoref1`. 

### Without `subfig` loaded:
screen shot 2018-10-17 at 1 44 43 pm

### With `subfig` loaded
screen shot 2018-10-17 at 1 45 18 pm

MWE at https://github.com/jcwhittier/maine-thesis/tree/subfig

File: Documentation/AppA.tex [hunk 3]
```diff
@@ -26,15 +26,7 @@ \section{Working Packages}
...
-\AfterPackage{subfig}{
-	\captionsetup{labelsep=period,listofformat=simple}
-	\renewcommand\p@figure{Figure\space}
-	\renewcommand\p@table{Table\space}
-}
-\end{verbatim}
-If you have defined additional floats, you will need more lines like those for figures and tables to your preamble.
+The one exception to this is if you use the caption or subfig package (which depends on the caption package) to create multi-page floats. The class file will correctly typeset the List of Figures/Tables, but you may have issues listing other floats. If you define additional floats, you will need more lines like those for figures and tables found in the class file to your preamble similar to the example:
 \begin{verbatim}
 \captionsetup{labelsep=period,listofformat=simple}
```
Comment (rpspringuel) [inline Documentation/AppA.tex:31]: I don't think this line is necessary here as it now exists in the AfterPackage block for the subfig package.

Comment (jcwhittier) [reply Documentation/AppA.tex:31]: Okay, I wasn't sure on that. I'm not sure if you saw, but I also noted an issue with referencing figures above. Would there be any way to enable the Figure and Table prefix only with their respective lists in the ToC?

**Comment by rpspringuel** [comment]
On the `\autoref` issue:  We might be able to fix this.  Two ideas come to mind: 

1. `\autoref` comes from `hyperref` so any patch would have to go in a `\AfterPackage` block that applies only when that package is loaded.  Since the error only occurs when both `subfig` and `hyperref` are loaded, I think that means nesting an `\AfterPackage{hyperref}` block inside the `\AfterPackage{subfig}` block.  I've never tried that before, though, so I don't know if it will work.

2. Instead of redefining `\p@figure` and `\p@table`, maybe there's a way to use the configuration capabilities of `subfig` to get the same effect in the LOF and LOT.  This isn't something I've tried before, but reading through the documentation seems to indicate that the `listofformat` option should deal with this.  Some experimentation would be required to see just what it does and if we can get it to do what we want.

**Comment by jcwhittier** [comment]
1. The issue is with both `\ref` and `\autoref`. With `\ref` 1.1 becomes Figure 1.1 and with `\autoref` Figure Figure 1.1. ` \cleveref` is also impacted.  Ideally ref would behave the same whether or not `subfig` were loaded.

2. This seems like the best option if feasible

**Comment by jcwhittier** [comment]
I was able to work around this issue by using the `subfigure` package instead of `subfig`

**Comment by rpspringuel** [comment]
That is supposed to be pointed out in the Documentation.  The only reason to use `subfig` over `subfigure` is if you need your floats to span multiple pages.  Perhaps the documentation on this point needs to be made clearer.

**Comment by rpspringuel** [comment]
Something to try: `\PassOptionsToPackage{caption=false}{caption}` instead of the `\AfterPackage{subfig}` block.  This should load the internals of the `caption` package (if it is requested), but not make any changes to the caption format.

**Comment by rpspringuel** [comment]
I've done some testing and `\PassOptionsToPackage{caption=false}{subfig}` instead of the `\AfterPackage{subfig}` block fixes the LOF and LOT problem without breaking `\ref` or `\autoref`.  I'm going to add that and update the documentation accordingly.

**Pull request closed without merging**